src/style.rs

```diff
@@ -0,0 +1,299 @@
+    ///
+    /// Fields overwritten by `egui::Style` are: `selection`, `background`, `tab_bar_background`, `tab_text`,
+    /// `tab_outline`, and `tab_background`.
+    pub fn from_egui(style: &egui::Style) -> Self {
+        Self {
+            selection_color: style.visuals.selection.bg_fill.linear_multiply(0.5),
+
+            background_color: style.visuals.window_fill(),
+            tab_bar_background: style.visuals.faint_bg_color,
+
+            separator_color: style.visuals.faint_bg_color,
+            border_color: style.visuals.faint_bg_color,
+            tab_text: style.visuals.widgets.active.fg_stroke.color,
+            tab_outline_color: style.visuals.widgets.active.bg_stroke.color,
+            tab_background_color: style.visuals.widgets.active.bg_fill,
+
+            ..Self::default()
+        }
+    }
+
+    pub(crate) fn hsplit(&self, ui: &mut Ui, fraction: &mut f32, rect: Rect) -> (Rect, Rect, Rect) {
+        let pixels_per_point = ui.ctx().pixels_per_point();
+
+        let mut separator = rect;
+
+        let midpoint = rect.min.x + rect.width() * *fraction;
+        separator.min.x = midpoint - self.separator_size * 0.5;
+        separator.max.x = midpoint + self.separator_size * 0.5;
+
+        let response = ui
+            .allocate_rect(separator, Sense::click_and_drag())
+            .on_hover_cursor(CursorIcon::ResizeHorizontal);
+
+        {
+            let delta = response.drag_delta().x;
+            let range = rect.max.x - rect.min.x;
+            let min = (self.separator_extra / range).min(1.0);
+            let max = 1.0 - min;
+            let (min, max) = (min.min(max), max.max(min));
+            *fraction = (*fraction + delta / range).clamp(min, max);
+        }
+
+        let midpoint = rect.min.x + rect.width() * *fraction;
+        separator.min.x = map_to_pixel(
+            midpoint - self.separator_size * 0.5,
+            pixels_per_point,
+            f32::round,
+        );
+        separator.max.x = map_to_pixel(
+            midpoint + self.separator_size * 0.5,
+            pixels_per_point,
+            f32::round,
+        );
+
+        (
+            rect.intersect(Rect::everything_right_of(separator.max.x)),
+            separator,
+            rect.intersect(Rect::everything_left_of(separator.min.x)),
+        )
+    }
+
+    pub(crate) fn vsplit(&self, ui: &mut Ui, fraction: &mut f32, rect: Rect) -> (Rect, Rect, Rect) {
+        let pixels_per_point = ui.ctx().pixels_per_point();
+
+        let mut separator = rect;
+
+        let midpoint = rect.min.y + rect.height() * *fraction;
+        separator.min.y = midpoint - self.separator_size * 0.5;
+        separator.max.y = midpoint + self.separator_size * 0.5;
+
+        let response = ui
+            .allocate_rect(separator, Sense::click_and_drag())
+            .on_hover_cursor(CursorIcon::ResizeVertical);
+
+        {
+            let delta = response.drag_delta().y;
+            let range = rect.max.y - rect.min.y;
+
```

Comment:
Delete the empty line (or add one after line 86 for consistency)

Reply:
Roger that.